figshare/figshare.py

```diff
@@ -290,3 +293,32 @@ def get_file_details(self, article_id, file_id):
         response = issue_request('GET', url,
                                  headers=self.get_headers(token=self.token))
         return response
+
+    def retrieve_files_from_article(self, article_id, directory=None):
+        """ Retrieve files and save them locally.
+
+        By default, files will be stored in the current working directory
+        under a folder called figshare_<article_id> by default.
+        Specify <outpath> for: <outpath>/figshare_<article_id>
+
+        Parameters
+        ----------
+        article_id : str or int
+            Figshare article ID
+
+        """
+
+        if directory is None:
+            directory = ''
+
+        # Get list of files
+        file_list = self.list_files(article_id)
+
+        n_files = len(file_list)
+        if n_files != 0:
+            dir0 = os.path.join(directory, "figshare_{0}/".format(article_id))
+            os.makedirs(dir0, exist_ok=True) # This might require Python >=3.2
+
+            for file_dict in file_list:
+               r = requests.get(file_dict['download_url']) #Will redirect
+               open(file_dict['name'], 'wb').write(r.content)
```

Comment:
does not need to be under the if statement.

Reply:
If statement entirely removed.